File: 0301_0400/0331/0331.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    h, m = (int(x) for x in f.readline().strip().split(':'))
    a, b = (int(x) for x in f.readline().split())
    h, m = ((h + a + 1) % 24, (m + b) % 60) if m + b >= 60 else ((h + a) % 24, m + b)
    q.write(f"{h:02d}:{m:02d}")

File: 0301_0400/0331/0331.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int h, m, a, b;

	fscanf(f, "%d:%d%d%d", &h, &m, &a, &b);
	h += a, m += b;
	if (m >= 60)
		m %= 60, ++h;
	h %= 24;
	fprintf(q, "%02d:%02d", h, m);
	return 0;
}
